Clean up LoggingTensor semantic

albanD opened the pull request:
Clarify how LoggingTensor works with autograd.
The updated comment should cover the semantic changes.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72623
* #72622
* __->__ #72620

Differential Revision: [D34214956](https://our.internmc.facebook.com/intern/diff/D34214956)

albanD committed:
Clean up LoggingTensor semantic

[ghstack-poisoned]

albanD commented on torch/testing/_internal/logging_tensor.py:
```diff
@@ -22,6 +22,13 @@ def no_dispatch() -> Iterator[None]:
 # 3. Enter dispatcher, wind your way through Autograd
 # 4. Hit Python dispatch key, call __torch_dispatch__
 
+# This Tensor can work with autograd in two ways:
```
Update this to say that allowing both is for testing only. And you should not mix the two in a single test or you're going to get very weird behavior!

zou3519 replied:
Question: are you actually planning on using LoggingTensor in both ways for testing?

albanD replied:
Yes. The current test do use the logging Tensor having gradients.
And the new ones added above in this stack check for the wrapped Tensor requirering gradients

zou3519 reviewed (approved):
LGTM. A note that mixing the two will save future test writers from a lot of confusion :)

albanD committed:
Update on "Clean up LoggingTensor semantic"

Clarify how LoggingTensor works with autograd.
The updated comment should cover the semantic changes.




[ghstack-poisoned]

albanD committed:
Update on "Clean up LoggingTensor semantic"

Clarify how LoggingTensor works with autograd.
The updated comment should cover the semantic changes.




[ghstack-poisoned]
